examples/jsm/lines/LineMaterial.js

```diff
@@ -243,159 +243,160 @@ ShaderLib[ 'line' ] = {
+
+					return this.uniforms.diffuse.value;
+
+				},
+
+				set: function ( value ) {
+
+					this.uniforms.diffuse.value = value;
+
+				}
 
 			},
 
-			set: function ( value ) {
+			linewidth: {
 
-				this.uniforms.diffuse.value = value;
+				enumerable: true,
 
-			}
+				get: function () {
 
-		},
+					return this.uniforms.linewidth.value;
 
-		linewidth: {
+				},
 
-			enumerable: true,
+				set: function ( value ) {
 
-			get: function () {
+					this.uniforms.linewidth.value = value;
 
-				return this.uniforms.linewidth.value;
+				}
 
 			},
 
-			set: function ( value ) {
+			dashScale: {
 
-				this.uniforms.linewidth.value = value;
+				enumerable: true,
 
-			}
+				get: function () {
 
-		},
+					return this.uniforms.dashScale.value;
 
-		dashScale: {
+				},
 
-			enumerable: true,
+				set: function ( value ) {
 
-			get: function () {
+					this.uniforms.dashScale.value = value;
 
-				return this.uniforms.dashScale.value;
+				}
 
 			},
 
-			set: function ( value ) {
+			dashSize: {
 
-				this.uniforms.dashScale.value = value;
+				enumerable: true,
 
-			}
+				get: function () {
 
-		},
+					return this.uniforms.dashSize.value;
 
-		dashSize: {
+				},
 
-			enumerable: true,
+				set: function ( value ) {
 
-			get: function () {
+					this.uniforms.dashSize.value = value;
 
-				return this.uniforms.dashSize.value;
+				}
 
 			},
 
-			set: function ( value ) {
+			gapSize: {
 
-				this.uniforms.dashSize.value = value;
+				enumerable: true,
 
-			}
+				get: function () {
 
-		},
+					return this.uniforms.gapSize.value;
 
-		gapSize: {
+				},
 
-			enumerable: true,
+				set: function ( value ) {
 
-			get: function () {
+					this.uniforms.gapSize.value = value;
 
-				return this.uniforms.gapSize.value;
+				}
 
 			},
 
-			set: function ( value ) {
+			opacity: {
 
-				this.uniforms.gapSize.value = value;
+				enumerable: true,
 
-			}
+				get: function () {
 
-		},
+					return this.uniforms.opacity.value;
 
-		opacity: {
+				},
 
-			enumerable: true,
+				set: function ( value ) {
 
-			get: function () {
+					this.uniforms.opacity.value = value;
 
-				return this.uniforms.opacity.value;
+				}
 
 			},
 
-			set: function ( value ) {
-
-				this.uniforms.opacity.value = value;
-
-			}
-
-		},
+			resolution: {
 
-		resolution: {
+				enumerable: true,
 
-			enumerable: true,
+				get: function () {
 
-			get: function () {
+					return this.uniforms.resolution.value;
 
-				return this.uniforms.resolution.value;
+				},
 
-			},
+				set: function ( value ) {
 
-			set: function ( value ) {
+					this.uniforms.resolution.value.copy( value );
 
-				this.uniforms.resolution.value.copy( value );
+				}
 
 			}
 
-		}
-
-	} );
+		} );
 
-	this.setValues( parameters );
+		this.setValues( parameters );
 
-};
+	}
 
-LineMaterial.prototype = Object.create( ShaderMaterial.prototype );
-LineMaterial.prototype.constructor = LineMaterial;
+}
 
 LineMaterial.prototype.isLineMaterial = true;
```

Comment:
could you use the `Object.defineProperty()` pattern demonstrated in #20110 across the files modified in this PR? 🙏

Also support for class fields is coming soon.

Reply:
OK.

Reply:
According to #21293, we should not use `Object.defineProperty()` anymore. So e.g.
```js
LineMaterial.prototype.isLineMaterial = true;
```
was actually fine.

Sorry for this back and forth!

Reply:
> Sorry for this back and forth!

It doesn't matter. It's not a big problem.